source/danger/danger_run.ts

```diff
@@ -36,31 +37,69 @@ export interface DangerRun extends RepresentationForURL {
 export const dangerRunForRules = (
   event: string,
   action: string | null,
-  rule: RunnerRuleset | undefined | null
+  rule: RunnerRuleset | undefined | null,
+  webhook: any
 ): DangerRun[] => {
   // tslint:disable-line
   // Can't do anything with nothing
   if (!rule) {
     return []
   }
 
+  // These are the potential keys that could trigger a run
   const directKey = event
   const globsKey = event + ".*"
   const dotActionKey = event + "." + action
 
-  const arraydVersions = Object.keys(rule)
+  const arrayVersions = Object.keys(rule)
+    // Look through all existing rules to see if we can
+    // find a key that matches the incoming rule
     .filter(key => {
-      const indvRules = key.split(",").map(i => i.trim())
+      // Take into account comma split strings:
+      //   "pull_request.opened, pull_request.closed"
+      //
+      // also take into account webhook checking:
+      //   "pull_request (pull_request.id == 12)"
+      //
+      const eachRule = key.split(",").map(i => i.split("(")[0].trim())
       const allKeys = [directKey, globsKey, dotActionKey]
-      return allKeys.some(key => indvRules.includes(key))
+      return allKeys.some(potentialKey => eachRule.includes(potentialKey))
+    })
+    .filter(key => {
+      // Do the webhook data check, return early if there's no () or ==
+      if (!key.includes("(") || !key.includes("==")) {
+        return true
+      }
```

Comment:
Forgive me if I'm misunderstanding the logic.

So from here on what happens if the key is:
`pull_request.updated, issue.labeled( issue.label.name == "question")`

If the keypath mapping fails and the event was really a `pull_request.updated`, isn't it skipped?

Reply:
This can be dealt with by only supporting one at once 👍